plugin/src/com/microsoft/alm/plugin/external/reactive/RdIdeaLoggerFactory.java

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root.
+
+package com.microsoft.alm.plugin.external.reactive;
+
+import com.jetbrains.rd.util.ILoggerFactory;
+import com.jetbrains.rd.util.LogLevel;
+import com.jetbrains.rd.util.Logger;
+import com.jetbrains.rd.util.Statics;
+import kotlin.jvm.JvmClassMappingKt;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class RdIdeaLoggerFactory implements ILoggerFactory {
```

Comment:
Oh, this is an intersting story. Look: by default, Rd's loggers use commons-logging library, and log everything (literally everything, with `TRACE` being enabled by default) into IDEA process' stdout.

In Rider, we override it and redirect everything into the IDEA's logging facility. So, in Rider, our plugin works well. But if I start it in IDEA, then it's starting to write traces of everything into stdout which doesn't work very well and performant.

After discussion with Dmitry Ivanov, I've come up with the current solution. IDEA **needs** that factory for the Rd logging to work properly.

I have verified that this doesn't interfer with Rd logging in Rider at all (because of how IDEA class loader works: we have a second copy of Rd framework embedded into the plugin, and any changes in the global Rd logger only affects the plugin, and doesn't interfer with Rider's Rd at all).